packages/next/src/server/config-shared.ts

```diff
@@ -605,6 +605,14 @@ export interface NextConfig extends Record<string, any> {
    */
   env?: Record<string, string | undefined>
 
+  /**
+   * Enable type-checking and autocompletion for environment variables.
+   *
+   * @default true
+   * @see https://nextjs.org/docs/app/api-reference/next-config-js/typedEnv
+   */
+  typedEnv?: boolean
```

Comment:
let's still put it along with the experimental flag, but default value is true

Reply:
That feels a bit weird tbh. Experimentals usually mean unstable. As they are unstable, why would they be on by default?

Reply:
It's not really experimental if we enable it by default anyway. The usual path would be experimental opt-in and then stable opt-out e.g. `config.experimental.enableTypedEnvironmentVariables` and then `config.disableTypedEnvironmentVariables`.

Though we should have a good idea why you'd want to disable it when it goes stable. Just adding options without explainers when you'd choose either value just contributes to decision paralysis.

Reply:
going experimental, default to false